internal/pkg/state/view.go

```diff
@@ -78,6 +80,20 @@ func (v *View) InitResolveAddress(ctx context.Context, a addr.Address) (addr.Add
 	return state.ResolveAddress(StoreFromCbor(ctx, v.ipldStore), a)
 }
 
+// Returns public key address if id address is given
+func (v *View) AccountSignerAddress(ctx context.Context, a addr.Address) (addr.Address, error) {
+	if a.Protocol() != addr.ID {
```

Comment:
Check for equality with SECP or BLS to avoid pass through of an actor temporary address.